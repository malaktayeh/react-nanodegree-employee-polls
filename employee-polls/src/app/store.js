import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import usersReducer from './features/usersSlice';
import authedUserReducer from './features/authedUserSlice';
import questionsReducer from './features/questionsSlice';

const reducer = combineReducers({
  users: usersReducer,
  authedUser: authedUserReducer,
  questions: questionsReducer
});

const store = configureStore({
  reducer
});

export default store;
