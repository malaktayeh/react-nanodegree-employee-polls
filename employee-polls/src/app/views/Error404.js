import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Container from 'react-bootstrap/esm/Container';
import Button from 'react-bootstrap/Button';
import NavBar from '../components/Navbar';
import { authedUserSelector } from '../features/authedUserSlice';

function Error404() {
  const { authedUser } = useSelector(authedUserSelector);

  return (
    <>
      {Object.keys(authedUser).length !== 0 ? <NavBar /> : null}
      <Container className="mt-5 mb-2">
        <div className="mt-5">
          <h2 className="mt-3"> 4 0 4 !</h2>
        </div>
        <Button variant="light" className="mt-5">
          <Link to="/" style={{ textDecoration: 'none', color: 'black' }}>
            {' '}
            ⬅️ Return{' '}
          </Link>
        </Button>
      </Container>
    </>
  );
}

export default Error404;
