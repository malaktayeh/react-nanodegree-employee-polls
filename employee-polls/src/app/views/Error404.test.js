import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../store';
import Error404 from './Error404';

describe('404 Page', () => {
  it('will match snapshot', () => {
    const view = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Error404 />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(view).toMatchSnapshot();
  });
});
