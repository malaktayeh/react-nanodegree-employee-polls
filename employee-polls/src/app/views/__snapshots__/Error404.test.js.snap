// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page will match snapshot 1`] = `
<div
  className="mt-5 mb-2 container"
>
  <div
    className="mt-5"
  >
    <h2
      className="mt-3"
    >
       4 0 4 !
    </h2>
  </div>
  <button
    className="mt-5 btn btn-light"
    disabled={false}
    type="button"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
    >
       
      ⬅️ Return
       
    </a>
  </button>
</div>
`;
